### DISCUSSION TITLE
Методы в PHP

### DISCUSSION THREAD

**Discussion by Nicolgeh** [totumonline/totum-ru-issues-and-discussions#499] [Как это сделать?] [answered]
В создании внешних функций имеется возможность обращаться к родительским методам различных классов, но интересуют определенные методы:
1. Получение ID таблицы. Как из кода PHP нам получить ID таблицы в которой срабатывает функция, а также всю инфу об этой таблице ( ID, nameTable, typeTable, cycleID, RowID, FieldName, FieldType )
2. Получение User'a который запустил выполнение скрипта ( инфа о юзере в тотуме )

**Comment by poluninatm** [comment, marked as answer]
Внешняя функция при запуске имеет в качестве $this соответствующий типу тотум-кода объект класса Calculate или его потомка.

Через `$this->Table` можно получить таблицу, в которой запущен код. `$this->Table->getTableRow()` содержит row с данными таблицы. `$this->Table->getCycle()?->getId()`  вернет  id цикла для расчетной таблицы в цикле или null для остальных

`$this->Table->getUser()` вернет объект класса User.

**Reply by Nicolgeh** [reply]
Это обращение внутри класса, но у меня функция не объединена в один класс, т.к. банально требуется создания нового объекта для авторизации для работы с таблицами ( insert, delete ), или инициализация класса который наследует `Conf` для того чтобы унаследовать `PROTECTED SmtpData `

**Reply by Nicolgeh** [reply]
Есть такой момент что при авторизации мы создаем новый объект `Totum`, и из него мы также можем обращаться ко всем методам Calculate, но в метод `getTableRow` необходимо в качестве параметра передать ID таблицы - что не подходит для моих задач, ведь надо вытащить ID таблицы из скрипта, а не передавать его в параметрах функции

**Reply by Nicolgeh** [reply]
Также метод `getUser` возвращает инфу о юзере от которого происходит авторизация в тотуме ( как и в доках это `anonym`), а не юзера из за которого сработал скрипт ( в моем случае триггер для запуска скрипта это кнопка )

**Reply by poluninatm** [reply]
Я поняла ваш вопрос, как вопрос о `ext*` функциям. Удачи в ваших изысканиях :)
